3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,77 @@
+//Voice assistant
+let personName = "";
+let toDoList = [];
+
+function getReply(command) {
+  if (command.startsWith("Hello my name is")) {
+    personName = command.split(" ")[4];
+    console.log(`nice to meet you ${personName}`);
+    return;
+  }
+
+  if (command === "What is my name") {
+    if (personName === "") console.log("I dont have your name");
+    else {
+      console.log(personName);
+    }
+    return;
+  }
+
+  if (command.startsWith("Add") && command.endsWith("to my todo")) {
+    // I did this part with for loop but i found easier way
+    //let commandArr = command.split(" ");
+    // for (let i = 1; i < commandArr.length - 3; i++) {
+    //   if (i == commandArr.length - 4)
+    // toDo += commandArr[i];
+    //   else toDo += commandArr[i] + " "; }
+
+    let toDo = "";
+    toDo = command.substring(4, command.length - 11);
+    toDoList.push(toDo);
+    console.log(`${toDo} added to your todo`);
+  }
+  if (command.startsWith("Remove")) {
```

Comment:
This `if` and others below will be checked even if the above case was hit. To make it easier to manage and less error prone, I'd suggest using `else if` syntax. The immediate benefit is that you do not need to return when a command has been processed and you do risk running multiple handlers for the same command; it is also marginally faster! 🚀